apps/music/js/music.js

```diff
@@ -160,88 +160,116 @@ var App = (function() {
     }
   }
 
-  function showCurrentView(callback) {
-    // We need AlbumArtCache.getCoverURL() to display thumbnails; it might not
-    // have been loaded yet, so make sure we load it first. This should prevent
-    // us from having to worry about loading it anywhere else in the code, since
-    // showCurrentView is called pretty early in the startup process.
-    LazyLoader.load('js/metadata/album_art_cache.js', function() {
-      function showListView() {
-        var option = TabBar.option;
-        var info = {
-          key: 'metadata.' + option,
-          range: null,
-          direction: (option === 'title') ? 'next' : 'nextunique',
-          option: option
-        };
-
-        ListView.activate(info);
+  function dbEnumerable(callback) {
+    // If we've been upgrading, hide that now
+    if (app.currentOverlay === 'upgrade') {
+      app.showOverlay(null);
+    }
+
+    // Display music that we already know about
+    refreshViews(function() {
+      // Tell performance monitors that the content is displayed and is
+      // ready to interact with. We won't send the final moz-app-loaded
+      // event until we're completely stable and have finished scanning.
+      //
+      // XXX: Maybe we could emit these events earlier, when we've just
+      // finished the "above the fold" content. That's hard to do on
+      // arbitrary screen resolutions, though.
+      window.performance.mark('visuallyLoaded');
+      window.dispatchEvent(new CustomEvent('moz-app-visually-complete'));
+      window.performance.mark('contentInteractive');
+      window.dispatchEvent(new CustomEvent('moz-content-interactive'));
```

Comment:
I've already reviewed this stuff in your other startup refactoring patch.  Looking at it now, I'll add that Eli is trying to get these dispatchEvent() calls out now.  You could help him out here if you take them out yourself, since his patch will have been completely broken by this.